test/e2e/storage/in_tree_volumes.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/drivers"
+	"k8s.io/kubernetes/test/e2e/storage/testpatterns"
+	"k8s.io/kubernetes/test/e2e/storage/testsuites"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+)
+
+var testDrivers = map[string]func() drivers.TestDriver{
+	"NFS":                        drivers.InitNFSDriver,
+	"GlusterFS":                  drivers.InitGlusterFSDriver,
+	"iSCSI [Feature:Volumes]":    drivers.InitISCSIDriver,
+	"Ceph-RBD [Feature:Volumes]": drivers.InitRbdDriver,
+	"CephFS [Feature:Volumes]":   drivers.InitCephFSDriver,
+	"hostpath":                   drivers.InitHostpathDriver,
+	"hostpathSymlink":            drivers.InitHostpathSymlinkDriver,
+	"emptydir":                   drivers.InitEmptydirDriver,
+	"cinder":                     drivers.InitCinderDriver,
+	"gce":                        drivers.InitGceDriver,
+	"vSphere":                    drivers.InitVSphereDriver,
+	"azure":                      drivers.InitAzureDriver,
+	"aws":                        drivers.InitAwsDriver,
+}
+
+var testSuites = []func() testsuites.TestSuite{
+	testsuites.InitVolumesTestSuite,
+	testsuites.InitVolumeIOTestSuite,
+	testsuites.InitVolumeModeTestSuite,
+	testsuites.InitSubPathTestSuite,
+}
+
+var _ = utils.SIGDescribe("In-tree Volumes", func() {
+	f := framework.NewDefaultFramework("volumes")
+
+	var (
+		cs     clientset.Interface
+		ns     *v1.Namespace
+		node   v1.Node
```

Comment:
Is this only needed for hostpath tests?  If so, can the hostpath driver setup do this?